src/MainWindow/main_window.cpp

```diff
@@ -147,6 +147,13 @@ void MainWindow::SetWindowTitle(const QString& filename, const QString& project,
   }
 }
 
+void MainWindow::CloseFileTabOpened() {
+  auto tabCount = TextEditorForm::Instance()->GetTabWidget()->count();
+  for (auto i = tabCount - 1; i >= 0; i--) {
```

Comment:
why not **for (int i = 0; i < tabCount; i++)** ? this is better for reading

Reply:
@KochynVolodymyr  @ravic-rs  The way I implement, When we remove any time it shift the tabs from right to left and index changes depending upon the number of tabs for example, If we have 3 tabs opened in the editor, Index would be 0, 1 2. Now if we removed tab on 0 index, tab on 1 will become index 0 as it will shift to place of 0 index. In our GUI tab shifts from right to left when they close. So it works good in this case.

If we go with suggested logic,  GUI crashes because So When we remove by index in each loop of above loop, it end up removing from index 0,1,2,3 etc, but after we've done  a remove the first time our list has changed.
So if we remove index 0 then everything shifts on index 1 becomes 0 etc, 2 becomes 1 etc, so on our next remove, we are removing at index 1 which is actually now pointing at 2. This continues each time the loop happens. So it crashes whenever tab count increase from more than 1.

Reply:
can you rework it like this:
`while (count != 0) {...}` ?